### PR TITLE
Salud mental encuentro fixes

### PR DESCRIPTION
**Pull request by mseaton**

### CODE REVIEW

File: jobs/sql/extractions/salud_mental_encuentro_staging.sql [hunk 1]
```diff
@@ -1028,383 +334,91 @@ SELECT concept_id INTO @obsessive_compulsive_disorder FROM concept WHERE uuid='1
...
-UPDATE salud_mental_encountero t 
-SET t.primary_diagnosis  = (
-SELECT diagnosis
-FROM diagnosis_pre_data dp
-WHERE dp.person_id=t.patient_id 
-AND dp.encounter_id=t.encounter_id 
-limit 1
-);
 
--- ---- Ascending Order ------------------------------------------
```
Comment (mseaton) [inline jobs/sql/extractions/salud_mental_encuentro_staging.sql:1323]: We should remove all of this ordering stuff I believe, in favor of doing this on the SQL Server side after the data has been loaded.  I removed it with that assumption.  @dmdesimone can you confirm?

File: jobs/sql/extractions/salud_mental_encuentro_staging.sql [hunk 2]
```diff
@@ -1028,383 +334,91 @@ SELECT concept_id INTO @obsessive_compulsive_disorder FROM concept WHERE uuid='1
...
+update salud_mental_encountero e inner join temp_obs o on e.encounter_id = o.encounter_id
+set e.primary_diagnosis = value_coded_name(o.obs_id, 'en')
+where o.concept_id = concept_from_mapping('PIH', '3064');
 
 -- --------------------------------------- Final Select -----------------------------------------------------------------------------------------------------------------------------------------
 SELECT 
 DISTINCT
-CONCAT(@partition,'-',emr_id) "emr_id" ,
+CONCAT(@partition,'-',emr_id) as emr_id,
 person_uuid,
```
Comment (mseaton) [inline jobs/sql/extractions/salud_mental_encuentro_staging.sql:410]: Using "as xxx" is considered better practice.

**Review by mseaton — COMMENTED** [review]
@dmdesimone / @aibrahem550 see PR and comments.  I'll attach to a ticket.

**Review by dmdesimone — APPROVED** [review]
Thanks @mseaton .  Looks good.  

Please take a look at this @aibrahem550

**Review by aibrahem550 — APPROVED** [review]

**Pull request merged**